### PR TITLE
Bug fix to allow access to top level params in reduce script

### PR DESCRIPTION
**Pull request by jdconrad**
Parameters specified for use in all scripts within a metric aggregation were never merged into reduce scripts as described in this issue (https://github.com/elastic/elasticsearch/issues/42046).  This fix merges top-level parameters for a metric aggregation into reduce script parameters.

### CODE REVIEW

**Comment by jdconrad** [comment]
@elasticmachine elasticsearch-ci/2

**Comment by jdconrad** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by jdconrad** [comment]
@elasticmachine run elasticsearch-ci/bwc

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/ScriptedMetricAggregatorTests.java [hunk 1]
```diff
@@ -72,6 +72,8 @@
             Collections.singletonMap("itemValue", 12));
     private static final Script COMBINE_SCRIPT_PARAMS = new Script(ScriptType.INLINE, MockScriptEngine.NAME, "combineScriptParams",
             Collections.singletonMap("divisor", 4));
+    private static final Script REDUCE_SCRIPT_AGG_PARAMS = new Script(ScriptType.INLINE, MockScriptEngine.NAME, "reduceScriptAggParams",
```
Comment (rjernst) [inline server/src/test/java/org/elasticsearch/search/aggregations/metrics/ScriptedMetricAggregatorTests.java]: Can this be named without "agg" like the similar constants here?

**Review by rjernst — APPROVED** [review]
LGTM

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/ScriptedMetricAggregatorTests.java (hunk 1 above)
Comment (jdconrad) [reply server/src/test/java/org/elasticsearch/search/aggregations/metrics/ScriptedMetricAggregatorTests.java]: Good catch.  Fixed.

**Comment by jdconrad** [comment]
@rjernst Thanks for the review!  Will commit as soon as CI passes.

**Comment by jdconrad** [comment]
@elasticmachine test this please

**Comment by jdconrad** [comment]
@elasticmachine run elasticsearch-ci/1

**Comment by jdconrad** [comment]
@elasticmachine run elasticsearch-ci/1

**Pull request merged**